### PR TITLE
*: adjust the project layout

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
This PR adjusts the current project layout and tries to make it well-organized.

### What is changed and how it works?
1. create `tools` directory under the root directory
2. move all binaries from `cmd` to `tools` except `pd-server`
3. rename `simulator` to `pd-simulator` in `tools`
4. rename `pkg/faketikv` directory to `pkg/simulator` directory and move it from `pkg` to `tools/pd-simulator`
5. move `pdctl` to `tools/pd-ctl`
6. add the `README.md` for each tool
7. rename `pd-client` to `client`

### Check List 

Tests 

 - No code

Related changes

 - Need to be included in the release notes

### CODE REVIEW

**Comment by disksing** [comment]
I think we should mention it in release notes because there are other projects use pd client as a library. They need to adapt their code when upgrade pd dependency.

**Comment by gregwebs** [comment]
Why cmd/ -> tools/? Note that retool creates a directory `_tools/` already which is currently gitignored.

**Comment by rleungx** [comment]
@gregwebs I noticed that. But the reason I do it is to separate the `pd-server` from the other tools. So I introduce `tools` directory like etcd does. Do you have any suggestion?

**Comment by gregwebs** [comment]
It does seem to be somewhat of a [standard](https://github.com/golang-standards/project-layout/tree/master/tools).

You could change the Makefile so that retool uses a different directory name by setting the `-tool-dir` option. However, I am less concerned about `_tools` and more about having a `tools.json` and a `tools/` directory. I don't think `tools.json` is configurable. It would be best to have `retool.json` and `_retool/`. I  opened a github issue for retool: https://github.com/twitchtv/retool/issues/48

**Comment by gregwebs** [comment]
@rleungx can you add `-tool-dir _retool` to the makefile (maybe we should have a `./hack/retool` that adds this) and also change the .gitignore? Renaming `tools.json` can come later when we hear back on that issue.

**Comment by gregwebs** [comment]
PR for a slight adjustment to the retools setup: https://github.com/rleungx/pd/pull/1

File: Makefile [hunk 1]
```diff
@@ -7,9 +7,10 @@ BASIC_TEST_PKGS := $(filter-out github.com/pingcap/pd/pkg/integration_test,$(TES
 PACKAGES := go list ./...
 PACKAGE_DIRECTORIES := $(PACKAGES) | sed 's|github.com/pingcap/pd/||'
 GOCHECKER := awk '{ print } END { if (NR > 0) { exit 1 } }'
+RETOOL:= ./hack/retool
 
-GOFAIL_ENABLE  := $$(find $$PWD/ -type d | grep -vE "(\.git|vendor)" | xargs retool do gofail enable)
-GOFAIL_DISABLE := $$(find $$PWD/ -type d | grep -vE "(\.git|vendor)" | xargs retool do gofail disable)
+GOFAIL_ENABLE  := $$(find $$PWD/ -type d | grep -vE "(\.git|vendor)" | xargs ./hack/retool do gofail enable)
+GOFAIL_DISABLE := $$(find $$PWD/ -type d | grep -vE "(\.git|vendor)" | xargs ./hack/retool  do gofail disable)
```
Comment (disksing) [inline Makefile]: I guess you have accidentally added a space.

**Comment by disksing** [comment]
@rleungx Please resolve the conflicts.

**Comment by rleungx** [comment]
PTAL. @disksing @nolouch

**Pull request merged**